float64[] goal
---
bool res
